store/postgres/tests/subgraph.rs

```diff
@@ -1060,3 +1067,85 @@ fn fail_unfail_non_deterministic_error_noop() {
         test_store::remove_subgraphs();
     })
 }
+
+#[test]
+#[serial]
+fn test_subgraph_migration_v2() {
+    const NAME: &str = "subgraph_manifest_v2";
+
+    async fn setup() -> DeploymentLocator {
+        let id = DeploymentHash::new(NAME).unwrap();
+        remove_subgraphs();
+        create_test_subgraph(&id, SUBGRAPH_GQL).await
+    }
+
+    let deployment = STORE_RUNTIME.handle().block_on(async { setup().await });
+    let deployment_schema = format!("sgd{}", &deployment.id);
+    run_test_with_conn(|conn| {
+        // change the schema version to v1 again.
+        let query = format!(
+            "UPDATE deployment_schemas SET version = 1 WHERE subgraph = '{}'",
+            NAME
+        );
+        conn.batch_execute(&query)
+            .expect("unable to change the schema version");
+
+        // change the datasource table as per v1.
+        let query = format!(
+            "
+        ALTER TABLE {}.data_sources$
+        DROP COLUMN done;
+        ",
+            deployment_schema
+        );
+        conn.batch_execute(&query)
+            .expect("Failed to drop test schema");
+    });
+
+    // run the migration for v2.
+    STORE_RUNTIME.handle().block_on(async {
+        SUBGRAPH_STORE.clear_caches();
+        SUBGRAPH_STORE.run_migration(&deployment.hash).unwrap();
+    });
+
+    // assert whether done column added to the datasource table.
+    run_test_with_conn(|conn| {
+        let query = format!(
+            "SELECT column_name
```

Comment:
Nice work on this test! Lets also select and test `is_nullable`, `column_default ` and if possible `data_type` here.